### PR TITLE
BOAC-2710: Adds created_by to authorized_users

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2710

### CODE REVIEW

File: boac/models/authorized_user.py [hunk 1]
```diff
@@ -55,42 +58,55 @@ class AuthorizedUser(Base):
...
         return f"""<AuthorizedUser {self.uid},
                     is_admin={self.is_admin},
                     in_demo_mode={self.in_demo_mode},
                     can_access_canvas_data={self.can_access_canvas_data},
-                    updated={self.updated_at},
-                    created={self.created_at},
-                    deleted={self.deleted_at}>
+                    created={self.created_at}{created_by},
+                    updated={self.updated_at}{updated_by},
+                    deleted={self.deleted_at}{deleted_by}>
```
Comment (johncrossman) [inline boac/models/authorized_user.py]: @lyttam Questions/comments on code above:
- why is `created_by` nullable?
- what is the reqt behind `updated_by` column?
- the `__repr__` above has ambiguous keys. And, did you intentionally exclude the `*_at` values?

File: boac/api/cache_utils.py [hunk 2]
```diff
@@ -178,6 +178,7 @@ def refresh_department_memberships():
                 continue
             user = AuthorizedUser.create_or_restore(
                 uid=membership['uid'],
+                created_by='boa',
```
Comment (johncrossman) [inline boac/api/cache_utils.py]: @lyttam - I assume `created_by` will have UIDs. In the spirit of type-consistency, maybe "boa" created users should have created_by='0'

**Comment by lyttam** [comment]
@johncrossman I axed `updated_by` and `deleted_by`, made `created_by` not nullable, and populated it with '2040' for all users.  When the nightly job runs, every `authorized_user` record it creates or restores will have `created_by` changed to '0'.

Once we're able to manually add/edit users via the admin page, `created_by` will be populated with the logged-in user's uid.

**Comment by lyttam** [comment]
Migration has been run on boac_dev.

**Comment by johncrossman** [comment]
Thank you @lyttam

**Pull request merged**